### PR TITLE
fix: Github OAuth Rehaul

### PR DESCRIPTION
**Pull request by Pranay105**
- Updated doc structure
- Grammar fixes

### CODE REVIEW

File: website/docs/getting-started/setup/instance-configuration/authentication/github-login.md [hunk 1]
```diff
@@ -9,61 +9,52 @@ sidebar_position: 2
 
 GitHub allows the integration of third-party applications using your GitHub Identity. Follow the below guide to configure GitHub OAuth.
```
Comment (jnikhila) [inline website/docs/getting-started/setup/instance-configuration/authentication/github-login.md:10]: Here we should highlight that Appsmith provides integration with GitHub.

File: website/docs/getting-started/setup/instance-configuration/authentication/github-login.md [hunk 2]
```diff
@@ -9,61 +9,52 @@ sidebar_position: 2
...
 
-![](</img/github_login.png>)
+## Configure the Github account
```
Comment (jnikhila) [inline website/docs/getting-started/setup/instance-configuration/authentication/github-login.md:12]: I think we can reword the heading to `Enable GitHub OAuth Integration`.

**Review by jnikhila — CHANGES_REQUESTED** [review]

File: website/docs/getting-started/setup/instance-configuration/authentication/github-login.md [hunk 3]
```diff
@@ -7,63 +7,54 @@ sidebar_position: 2
...
 :::
+There are two ways to configure the self-hosted Appsmith instance - 
+- Admin Settings
+- Environment Variables
+
+**Admin Settings:** you can add the Client ID, Client Secret and allowed domains from [Admin Settings](/getting-started/setup/instance-configuration/). Go to *Admin Settings> Authentication> Github*, fill the required fields and click the **Save & Restart** button at the bottom of the page.
+
+![](/img/as_github_auth_config.png)
+
+**Environment Variables:** Navigate to the `docker.env` file located in `<PROJECT_ROOT>/stacks/configuration/`. Update the values for the following keys:
```
Comment (danciaclara) [inline website/docs/getting-started/setup/instance-configuration/authentication/github-login.md:45]: check how we did for Google. don't write for only Docker. 
Screenshot 2023-03-14 at 13 34 15

File: website/docs/getting-started/setup/instance-configuration/authentication/github-login.md [hunk 4]
```diff
@@ -7,63 +7,54 @@ sidebar_position: 2
...
 
 :::caution
-The Client ID and Client Secret entered aren't verified by any means. Please test it out manually before disabling the Form login authentication method. If Form login is disabled before testing the GitHub authentication configuration setup, it may lead to the user getting stuck on the login screen if the GitHub configuration is invalid. You’ll have to update the environment variables manually to fix this, as you can't update it via the Admin settings page since you won’t be logged in anymore
+Test the Client ID and Client Secret before disabling Form login on Appsmith. This ensures that users can access Appsmith with their Github credentials. To fix an invalid Github configuration, update the environment variables.
 :::
+There are two ways to configure the self-hosted Appsmith instance - 
+- Admin Settings
+- Environment Variables
+
+**Admin Settings:** you can add the Client ID, Client Secret and allowed domains from [Admin Settings](/getting-started/setup/instance-configuration/). Go to *Admin Settings> Authentication> Github*, fill the required fields and click the **Save & Restart** button at the bottom of the page.
```
Comment (danciaclara) [inline website/docs/getting-started/setup/instance-configuration/authentication/github-login.md:41]: https://developers.google.com/style/procedures

```suggestion
**Admin Settings:** Click **Admin Settings** > **Authentication** > **Enable** (Github). Add the Client ID and  Client Secret. Click the **Save & Restart** button at the bottom of the page. 
```

**Review by danciaclara — CHANGES_REQUESTED** [review]
For images, crop images or only grab a screenshot of the immediate area so we can  see them clearly.  Also for Github we only need to show the "Register a new OAuth Application" page
Screenshot 2023-03-14 at 13 29 29

File: website/docs/getting-started/setup/instance-configuration/authentication/github-login.md (hunk 3 above)
Comment (Pranay105) [reply website/docs/getting-started/setup/instance-configuration/authentication/github-login.md:45]: We need to tell the users where to find the config file. Just saying update the configuration file feels like half of the information. How do we solve this situation?

File: website/docs/getting-started/setup/instance-configuration/authentication/github-login.md [hunk 5]
```diff
@@ -7,63 +7,54 @@ sidebar_position: 2
...
-GitHub allows the integration of third-party applications using your GitHub Identity. Follow the below guide to configure GitHub OAuth.
+Appsmith allows you to integrate with Google OAuth 2.0, enabling end users to sign in to the Appsmith account using their Github authentication credentials.
 
-![](</img/github_login.png>)
+## Enable GitHub OAuth Integration
 
-To enable GitHub Sign in, login to your [GitHub Account](https://github.com)
+To enable GitHub Sign in, login to your [GitHub Account](https://github.com) and follow the steps below:
 
-**1. Navigate to the** [**Organizations section**](https://github.com/settings/organizations) **of your profile and the settings of the organization under which you want to create an OAuth App**
```
[the remark this answers is not in the document]
Comment (Pranay105) [reply website/docs/getting-started/setup/instance-configuration/authentication/github-login.md:16]: The steps I have given are valid for a specific organization. This is for a global OAuth App for a profile. I am updating the instructions.

File: website/docs/getting-started/setup/instance-configuration/authentication/github-login.md [hunk 6]
```diff
@@ -7,63 +7,54 @@ sidebar_position: 2
 
 # GitHub OAuth
 
-GitHub allows the integration of third-party applications using your GitHub Identity. Follow the below guide to configure GitHub OAuth.
+Appsmith allows you to integrate with Google OAuth 2.0, enabling end users to sign in to the Appsmith account using their Github authentication credentials.
```
[the remark this answers is not in the document]
Comment (Pranay105) [reply website/docs/getting-started/setup/instance-configuration/authentication/github-login.md:10]: Kept this image size to show the "Save and Restart" button.

[the remark this answers is not in the document]
Comment (danciaclara) [reply website/docs/getting-started/setup/instance-configuration/authentication/github-login.md]: hmm, ok got it.

File: website/docs/getting-started/setup/instance-configuration/authentication/github-login.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (Pranay105) [reply website/docs/getting-started/setup/instance-configuration/authentication/github-login.md:16]: I don't know the reason. It was previously mentioned for a specific organization. I followed the same. Will confirm to from the engg. team

File: website/docs/getting-started/setup/instance-configuration/authentication/github-login.md (hunk 3 above)
Comment (danciaclara) [reply website/docs/getting-started/setup/instance-configuration/authentication/github-login.md:41]: Kubernetes, AWS,  Docker have different configuration files. I don't know if we will be able to address that here. But you can say something like 

Update the values for the following keys in the instance configuration file, for example, in the `docker.env` file for Docker installation and in the `values.yaml` file for Kubernetes.

File: website/docs/getting-started/setup/instance-configuration/authentication/github-login.md [hunk 7]
```diff
@@ -7,63 +7,50 @@ sidebar_position: 2
...
+  # Example configuration in docker.env file
+  # ********* GitHub OAuth **********
+  APPSMITH_OAUTH2_GITHUB_CLIENT_ID=YOUR_GITHUB_CLIENT_ID
+  APPSMITH_OAUTH2_GITHUB_CLIENT_SECRET=YOUR_GITHUB_CLIENT_SECRET
+  APPSMITH_SIGNUP_ALLOWED_DOMAINS=exampledomain.com
+  # ******************************
+   ```
+* Restart the Appsmith instance. 
+
+After these steps, your Appsmith installation should now enable GitHub Login. If you are running appsmith behind an ELB / Proxy, please configure Nginx [following this guide](/help-and-support/troubleshooting-guide/deployment-errors#oauth-sign-up-not-working).
```
Comment (danciaclara) [inline website/docs/getting-started/setup/instance-configuration/authentication/github-login.md:56]: The ELB/Proxy piece appears to be a Troubleshooting guide

File: website/docs/getting-started/setup/instance-configuration/authentication/github-login.md [hunk 8]
```diff
@@ -7,63 +7,50 @@ sidebar_position: 2
...
-**1. Navigate to the** [**Organizations section**](https://github.com/settings/organizations) **of your profile and the settings of the organization under which you want to create an OAuth App**
-
-![Click to expand](</img/Github_Orgs.png>)
-
-**2. Navigate to the OAuth Apps section of the organization settings page**
-
-![Click to expand](</img/Github_OAuth_Apps.png>)
-
-**3. Create a new OAuth App**
+1. In your profile drop-down, navigate to **Settings > Developer Settings > OAuth Apps** and click on Register an Application to create and register a new OAuth App.
```
Comment (danciaclara) [inline website/docs/getting-started/setup/instance-configuration/authentication/github-login.md:16]:
```suggestion
1. In your profile drop-down, navigate to **Settings > Developer Settings > OAuth Apps** and click on **Register a new application** to create and register a new OAuth App.
```

**Review by danciaclara — CHANGES_REQUESTED** [review]

File: website/docs/getting-started/setup/instance-configuration/authentication/github-login.md (hunk 3 above)
Comment (danciaclara) [reply website/docs/getting-started/setup/instance-configuration/authentication/github-login.md:41]: Hey did you add this? Can't see it in the PR

File: website/docs/getting-started/setup/instance-configuration/authentication/github-login.md (hunk 7 above)
Comment (Pranay105) [reply website/docs/getting-started/setup/instance-configuration/authentication/github-login.md:56]: Yes, it is. We kept it like this as it was a specific error.

File: website/docs/getting-started/setup/instance-configuration/authentication/github-login.md (hunk 3 above)
Comment (Pranay105) [reply website/docs/getting-started/setup/instance-configuration/authentication/github-login.md]: Added it in google oauth. The changes in GitHub oauth file didn't get committed for some reason. Added it now.

**Review by danciaclara — DISMISSED** [review]

**Review by jnikhila — APPROVED** [review]

**Review by jnikhila — APPROVED** [review]

**Pull request merged**